src/GenericConjugacyClasses.jl

```diff
@@ -0,0 +1,499 @@
+import Oscar: norm, scalar_product, class_multiplication_coefficient, order, number_of_conjugacy_classes
+
+abstract type AbstractGenericConjugacyClass end
+
+@doc raw"""
+    GenericConjugacyClass <: AbstractGenericConjugacyClass
+
+The type for generic conjugacy classes. These are the generic conjugacy classes used in `CharTable`.
+"""
+struct GenericConjugacyClass <: AbstractGenericConjugacyClass
+	parent::CharTable
+	index::Int64
+	values::Union{Vector{GenericCyclo}, Nothing}
+	substitutions::Vector{ParameterSubstitution}
+end
+
+@doc raw"""
+    conjugacy_class_type(t::CharTable, class::Int64)
+
+Return the generic conjugacy class `class` of `t`.
+"""
+conjugacy_class_type(t::CharTable, class::Int64) = GenericConjugacyClass(t, class, nothing, ParameterSubstitution[])
+
+getindex(c::GenericConjugacyClass, i::Integer) = c.values === nothing ? parent(c)[i][c.index] : c.values[i]
+
+eltype(::Type{GenericConjugacyClass}) = GenericCyclo
+
+@doc raw"""
+    SimpleGenericConjugacyClass <: AbstractGenericConjugacyClass
+
+The type for simple generic conjugacy classes. These are the generic conjugacy classes used in `SimpleCharTable`.
+"""
+struct SimpleGenericConjugacyClass{T} <: AbstractGenericConjugacyClass
+	parent::SimpleCharTable{T}
+	index::Int64
+end
+
+@doc raw"""
+    conjugacy_class_type(t::SimpleCharTable{T}, class::Int64) where T <: NfPoly
+
+Return the generic conjugacy class `class` of `t`.
+"""
+conjugacy_class_type(t::SimpleCharTable{T}, class::Int64) where T <: NfPoly = SimpleGenericConjugacyClass{T}(t, class)
+
+getindex(c::SimpleGenericConjugacyClass, i::Integer) = parent(c)[i][c.index]
+
+eltype(::Type{SimpleGenericConjugacyClass{T}}) where T<:NfPoly = T
+
+AbstractAlgebra.parent(c::AbstractGenericConjugacyClass) = c.parent
+
+length(c::AbstractGenericConjugacyClass) = length(parent(c))
+iterate(c::AbstractGenericConjugacyClass, state::Integer=1) = state > length(c) ? nothing : (c[state], state+1)
+
+@doc raw"""
+    info(class::AbstractGenericConjugacyClass)
+
+Return the infolists of the generic conjugacy class `class`.
+"""
+info(class::AbstractGenericConjugacyClass) = parent(class).classinfo[class.index]
+
+@doc raw"""
+    info(t::Table, class::Int64)
+
+Return the infolists of the class type `class` of the table `t`.
```

Comment:
TODO: extend this to mention that the meaning of the content of these infolists differs between tables and the table info string usually contains more info about the format/content.